firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteSchema.java

```diff
@@ -351,6 +355,16 @@ private void addSequenceNumber() {
     }
   }
 
+  private void addUpdateTime() {
+    if (!tableContainsColumn("remote_documents", "snapshot_version_seconds")) {
+      hardAssert(
+          !tableContainsColumn("remote_documents", "snapshot_version_nanos"),
+          "Table contained snapshot_version_seconds, but is missing snapshot_version_nanos");
+      db.execSQL("ALTER TABLE remote_documents ADD COLUMN snapshot_version_seconds INTEGER");
+      db.execSQL("ALTER TABLE remote_documents ADD COLUMN snapshot_version_nanos INTEGER");
+    }
```

Comment:
If all the documents in the cache have a zero version doesn't that mean the query to find recently updated documents has to also check for zero? Doesn't that nearly nullify the benefits of index-free on any device that has pre-existing data?

Consider an existing device containing two targets, A updated at T1 matching D1 and D2. There's also a second target, listened to later: B updated at T2 matching D3. D3 actually matches the query for target A. All of this happened before this schema upgrade, so after the upgrade they'll all have update_time of zero.

If the user queries on A again we need to find D3, but it's going to have a zero update_time.

I think we're better off either:
  * actually populating update_time based on the value from inside the document.
  * dropping the contents of the remote_documents table

Though the latter is pretty unfriendly so I think we should only consider it if all other possibilities are exhausted.

Alternatively, maybe you have something up your sleeve to handle this case that I'm not seeing :-).

Reply:
I tried to describe my strategy for this in the PR comment, but added it as a comment in the code itself for more visibility. Basically - I haven't quite decided what the best way forward it as migrating the contents of the entire RemoteDocumentCache could be quite slow. I decided to punt on this till later (when we actually want to enable this).